frontends/common/constantFolding.h

```diff
@@ -23,36 +23,81 @@ limitations under the License.
 #include "frontends/p4/typeChecking/typeChecker.h"
 
 namespace P4 {
-// Can also be run before type checking, so it only uses types if
-// they are available.
+
+/** Implements a pass that statically evaluates many constant
+ * expressions.
+ *
+ * The pass can be invoked either with or without the reference and
+ * type maps. When type information is not available, constant folding
+ * is not done for some IR nodes.
+ *
+ * @pre: None
+ *
+ * @post: Ensures that
+ *    - some constant expressions are eliminated (TODO: specify?)
```

Comment:
I would rather say that some expressions that provably evaluate to a constant are replaced with the constant value.